File: vuln-sandbox/ruby/cwe-79/xss.rb
# CWE-79: Cross-Site Scripting via html_safe in Ruby/Rails
# Rule: ruby-xss-html-safe
# Expected: TruePositive

class CommentsController < ApplicationController
  def show
    # VULNERABLE: User input marked as html_safe bypasses Rails XSS protection
    @comment = params[:comment].html_safe
    render inline: "<p><%= @comment %></p>"
  end

  def preview
    # VULNERABLE: Request parameter marked as html_safe
    content = request.params[:content].html_safe
    render html: content
  end
end

File: vuln-sandbox/php/cwe-79/xss.php
<?php
// CWE-79: Cross-Site Scripting via echo with unescaped user input
// Rule: php-xss-echo-user-input
// Expected: TruePositive

// VULNERABLE: Echoing GET parameter without encoding
echo $_GET['name'];

// VULNERABLE: Echoing POST data directly
echo $_POST['message'];

// VULNERABLE: Echoing REQUEST data
echo $_REQUEST['search'];
?>
